akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -206,7 +206,27 @@ case object Lookup {
 
   private val SrvQuery = """^_(.+?)\._(.+?)\.(.+?)$""".r
 
-  private val DomainName = "^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$".r
+  /**
+   * Validates domain name:
```

Comment:
So these rules come from https://tools.ietf.org/html/rfc1034 ? Perhaps link to that here?